datatransfer/impl/graphsync/graphsync_impl.go

```diff
@@ -266,35 +279,14 @@ func (impl *graphsyncImpl) InProgressChannels() map[datatransfer.ChannelID]datat
 	return channelsCopy
 }
 
-// hasPushChannel returns true if a channel with ID chid exists and is for a Push request.
-func (impl *graphsyncImpl) hasPushChannel(chid datatransfer.ChannelID) bool {
-	return impl.getPushChannel(chid) != datatransfer.EmptyChannelState
-}
-
-// hasPullChannel returns true if a channel with ID chid exists and is for a Pull request.
-func (impl *graphsyncImpl) hasPullChannel(chid datatransfer.ChannelID) bool {
-	return impl.getPullChannel(chid) != datatransfer.EmptyChannelState
-}
-
-// getPullChannel searches for a pull-type channel in the slice of channels with id `chid`.
+// getChannel searches for a pull-type channel in the slice of channels with id `chid`.
 // Returns datatransfer.EmptyChannelState if:
 //   * there is no channel with that id
-//   * it is not related to a pull request
-func (impl *graphsyncImpl) getPullChannel(chid datatransfer.ChannelID) datatransfer.ChannelState {
+func (impl *graphsyncImpl) getChannel(chid datatransfer.ChannelID, sender peer.ID) datatransfer.ChannelState {
```

Comment:
I might rename this getChannelByIdAndSender -- or getMatchingChannel -- just cause it's slightly unclear why you need an chID & a sender for just looking up a channel -- when what you really mean here is:
- look up the channel matching this channel id
- but only return it if the sender matches as well (presumably to distinguish push/pull)

Reply:
I very nearly called it that and decided to keep the name short, but I'll change it.